src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -1,186 +1,346 @@
+			else
+			{
+				return CoreDispatcher.Main.RunWithResultAsync(
+					priority: CoreDispatcherPriority.Normal,
+					task: () => GetGeopositionAsync()
+				);
+			}
+		}
+
+
+		public static async Task<GeolocationAccessStatus> RequestAccessAsync()
+		{
+			var status = GeolocationAccessStatus.Allowed;
+
+			if (!await PermissionsHelper.CheckFineLocationPermission(CancellationToken.None))
+			{
+				status = await PermissionsHelper.TryGetFineLocationPermission(CancellationToken.None)
+					? GeolocationAccessStatus.Allowed
+					: GeolocationAccessStatus.Denied;
+
+				BroadcastStatus(PositionStatus.Initializing);
+				if (status == GeolocationAccessStatus.Allowed)
+				{
+					BroadcastStatus(PositionStatus.Ready);
+
+					// If geolocators subscribed to PositionChanged before the location permission was granted,
+					// make sure to initialize these geolocators now so they can start broadcasting.
+					foreach (var subscriber in _positionChangedSubscriptions)
+					{
+						subscriber.Key.TryInitialize();
+					}
+				}
+				else
+				{
+					BroadcastStatus(PositionStatus.Disabled);
+
+					foreach (var subscriber in _positionChangedSubscriptions)
+					{
+						subscriber.Key.WaitForPermissionFromBackground();
+					}
+				}
+			}
+
+			return status;
+		}
+
+		public void WaitForPermissionFromBackground()
+		{
+			CoreApplication.Resuming -= CoreApplication_Resuming;
+			CoreApplication.Resuming += CoreApplication_Resuming;
+		}
+
+		private void CoreApplication_Resuming(object sender, object e)
+		{
+			CoreDispatcher.Main.RunAsync(
+				priority: CoreDispatcherPriority.Normal,
+				handler: InitializeIfPermissionIsGranted
+			);
+		}
+
+		private async void InitializeIfPermissionIsGranted()
+		{
+			// If the user has granted the location permission while the app was in background, Initialize
+			if (await PermissionsHelper.CheckFineLocationPermission(CancellationToken.None))
+			{
+				TryInitialize();
+				CoreApplication.Resuming -= CoreApplication_Resuming;
+			}
+		}
+
+		private LocationManager InitializeLocationProvider(double desiredAccuracy)
+		{
+			var locationManager = (LocationManager)global::Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
+
+			var criteriaForLocationService = new Criteria
+			{
+				Accuracy = Accuracy.Coarse
+			};
+
+			var acceptableLocationProviders = locationManager.GetProviders(criteriaForLocationService, true);
+
+			if (acceptableLocationProviders.Any())
+			{
+				_locationProvider = acceptableLocationProviders.First();
+			}
+			else
+			{
+				_locationProvider = String.Empty;
+			}
+
+			return locationManager;
+		}
+
+		partial void StartPositionChanged()
+		{
+			_positionChangedSubscriptions.TryAdd(this, 0);
+		}
+
+		partial void StopPositionChanged()
+		{
+			_positionChangedSubscriptions.TryRemove(this, out var _);
+		}
+
+		public async Task<Windows.Devices.Geolocation.Geoposition> GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)

```

Comment:
The whole method is async, the body should be executed in an Task wrapper as well (inner functions will likely block).

Reply:
In first version of this PR, I have GetGeopositionAsync() redirected to `GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)` with default values of these two parameters (timeout 60 secs).
But then, Geolocator implementation was drastically changed, in a way I can't really merge changes. So, I choose to use my implementation for `GetGeopositionAsync` with parameters, and left parameterless `GetGeopositionAsync()` without change.

Reply:
Sure. It'll be best to adjust both methods so they have the same behavior, with default values.

Reply:
Yes. But as I wrote, "Geolocator implementation was drastically changed, in a way I can't really merge changes" (commits from Sep 8..14, 2020 by jcantin-nventive) (or, it would take too much of my time).

Reply:
Can I get any reply?

Reply:
Can I treat "no reply" as "agreed"?